dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchServiceCollectionExtensions.cs

```diff
@@ -20,229 +20,358 @@ namespace Microsoft.SemanticKernel;
 public static class AzureAISearchServiceCollectionExtensions
 {
     /// <summary>
-    /// Register an Azure AI Search <see cref="IVectorStore"/> with the specified service ID and where <see cref="SearchIndexClient"/> is retrieved from the dependency injection container.
+    /// Registers an Azure AI Search <see cref="IVectorStore"/> in the <see cref="IServiceCollection"/>, retrieving the <see cref="SearchIndexClient"/> from the dependency injection container.
     /// </summary>
-    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="IVectorStore"/> on.</param>
-    /// <param name="options">Optional options to further configure the <see cref="IVectorStore"/>.</param>
-    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    /// <param name="serviceCollection">The <see cref="IServiceCollection"/> to which the vector store should be added.</param>
+    /// <param name="options">Options to further configure the <see cref="IVectorStore"/>.</param>
+    /// <param name="lifetime">The service lifetime for the store. Defaults to <see cref="ServiceLifetime.Transient"/>.</param>
```

Comment:
Why `Transient` is the default? I would hope Azure AI Search SDK client to be the same as other Azure SDK clients:

https://devblogs.microsoft.com/azure-sdk/lifetime-management-and-thread-safety-guarantees-of-azure-sdk-net-clients/#client-lifetime

Reply:
Yep, I agree - see https://github.com/microsoft/semantic-kernel/issues/10549#issuecomment-2720619607 and below for the discussion. I'm mainly saving this for a later PR as it's a more important behavioral change, and I didn't want to do it all in one PR.